### PR TITLE
Perfomance metrics refinement

### PR DESCRIPTION
**Pull request by ludv1x**
METR-19372
METR-23305

### CODE REVIEW

File: dbms/src/Interpreters/executeQuery.cpp [hunk 1]
```diff
@@ -191,13 +193,23 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
...
 		}
 
+		if (res.out)
+		{
+			if (auto stream = dynamic_cast<CountingBlockOutputStream *>(res.out.get()))
+			{
+				stream->setProcessListElement(context.getProcessListElement());
+			}
+		}
+		if (!res.out || !dynamic_cast<CountingBlockOutputStream *>(res.out.get()))
```
Comment (alexey-milovidov) [inline dbms/src/Interpreters/executeQuery.cpp]: This is not needed right now?
(Misleading message for every non-INSERT query.)

File: dbms/src/Storages/MergeTree/MergeTreeDataMerger.cpp [hunk 2]
```diff
@@ -386,38 +387,41 @@ class ColumnSizeEstimator
...
-		}
-		else
-		{
-			merge_entry->progress = average_elem_progress * merge_entry->rows_with_key_columns_read;
-		}
+		merge_entry->bytes_read_uncompressed += value.bytes;
+		merge_entry->rows_read += value.rows;
+		merge_entry->progress = average_elem_progress * merge_entry->rows_read;
 	};
 };
```
Comment (alexey-milovidov) [inline dbms/src/Storages/MergeTree/MergeTreeDataMerger.cpp]: Missing comment about this class.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]

File: dbms/include/DB/Interpreters/QueryLog.h [hunk 3]
```diff
@@ -36,6 +35,11 @@ struct QueryLogElement
 	UInt64 read_rows{};
 	UInt64 read_bytes{};
 
+	UInt64 written_rows{};
+	UInt64 written_bytes{};
+
+	/// NOTE: Not obvious metric.
```
Comment (ludv1x) [inline dbms/include/DB/Interpreters/QueryLog.h]: So, as I metioned on next line, result_rows will be equal to written_rows (?)

**Comment by ludv1x** [comment]
Updated

File: doc/benchmark.html [hunk 4]
```diff
@@ -1555,13 +1555,13 @@
     html += "<tr>";
     html += "<th></th>";
     html += "<th style='text-align: left;'>Query</th>";
-    for (j in filtered_results) {
+    for (j = 0; j < filtered_results.length; j++) {
```
Comment (alexey-milovidov) [inline doc/benchmark.html]: No `var`?

File: CMakeLists.txt [hunk 5]
```diff
@@ -46,30 +46,30 @@ set (COMMON_WARNING_FLAGS "-Wall -Werror")
 set (CXX_WARNING_FLAGS "-Wnon-virtual-dtor -Wold-style-cast")
 
 set (CXX11_ABI "DEFAULT" CACHE STRING "Use C++11 ABI: DEFAULT, ENABLE, DISABLE")
-set (TEST_COVERAGE FALSE CACHE BOOL "Enables flags for test coverage")
-set (ENABLE_TESTS TRUE CACHE BOOL "Enables tests")
+option (TEST_COVERAGE "Enables flags for test coverage" OFF)
```
Comment (alexey-milovidov) [inline CMakeLists.txt]: Need instruction on how to install newer CMake in build.md.

File: doc/benchmark.html (hunk 4 above)
Comment (ludv1x) [inline doc/benchmark.html]: It seems that both variant work properly in this context. But local `var` is really better.

File: CMakeLists.txt (hunk 5 above)
Comment (ludv1x) [inline CMakeLists.txt]: Do you mean describe behavior of additional options?
Default behavior remains the same, and scripts from `build.md` seem to be workable.

Comment (ludv1x) [inline CMakeLists.txt]: Let merge this PR containing mixture of different changes and refine separate changes lately.

Comment (alexey-milovidov) [inline CMakeLists.txt]: Ok.
What I mean:

I have 
```
$ cmake --version
cmake version 2.8.12.2
```

and when I try to install newer CMake from sources,
I just get immediate segfault on cmake invocation.

**Pull request merged**